### PR TITLE
Opponents Shaymin Sky form correctly reverts on frozen status

### LINKED ISSUE #4539: Shaymin Form change from Sky to Land with Incorrect Status
### Description

Shaymin-Sky is supposed to change to Shaymin-Land when it becomes frozen or frostbitten. I am seeing it revert to Shaymin-Land with both Toxic poison and Burn (have not tested behavior with all other statuses).
Relevant code in `form_change_tables.h`:

```
#if P_FAMILY_SHAYMIN
static const struct FormChange sShayminFormChangeTable[] = {
    {FORM_CHANGE_ITEM_USE,    SPECIES_SHAYMIN_SKY, ITEM_GRACIDEA, DAY},
    {FORM_CHANGE_WITHDRAW,    SPECIES_SHAYMIN_LAND},
    {FORM_CHANGE_TIME_OF_DAY, SPECIES_SHAYMIN_LAND, NIGHT},
    {FORM_CHANGE_STATUS,      SPECIES_SHAYMIN_LAND, (STATUS1_FREEZE | STATUS1_FROSTBITE)},
    {FORM_CHANGE_TERMINATOR},
};
#endif //P_FAMILY_SHAYMIN
```

shayminformbug

### Version

1.8.1

### Upcoming/master Version

_No response_

### Discord contact info

iriv24#8479

### PR DESCRIPTION
**Pull request by TheJamesLJ**
## Description

Shaymin Sky would revert when receiving any status condition, additionally if the status condition was inflicted from a damaging move, after reverting form, the attacks damage would be applied again.
Shaymin_Sky_Land_Form_Change_Paralyze_Dmg

I have updated `GetFormChangeTargetSpeciesBoxMon` to check if the mons status matches the status on the form change table.

I have also updated the `BattleScript_TargetFormChangeNoPopup` battle script which checks `.if B_DISGUISE_HP_LOSS >= GEN_8` to apply HP loss after a disguise break, I believe this is intended for Mimikyu's disguise ability, so I added a new battle script function that applies the intended HP loss if the targets ability is `ABILITY_DISGUISE`.

## Images

Shaymin Sky correctly reverting form when receiving the Frozen status, and not receiving additional damage.
Shaymin_Sky_Land_Form_Change_Frozen

Shaymin Sky no longer changing form on other statuses (Poison & Burn).
Shaymin_Sky_Form_Poison
(Sorry, I was using emulator speed up which skipped the Fire Punch anim)
Shaymin_Sky_Form_Burn

Mimikyu disguise break still functioning as intended.
Mimikyu_Disguise_Break_HP_Drop

## Issue(s) that this PR fixes

Fixes #4539.

## **Discord contact info**

JamesLJ#5168

### CODE REVIEW

**Comment by AlexOn1ine** [comment]
Just in case someone wants to merge, I'll try to write a couple of tests later

File: src/pokemon.c [hunk 1]
```diff
@@ -6230,9 +6230,12 @@ u16 GetFormChangeTargetSpeciesBoxMon(struct BoxPokemon *boxMon, u16 method, u32
                     break;
                 case FORM_CHANGE_WITHDRAW:
                 case FORM_CHANGE_FAINT:
-                case FORM_CHANGE_STATUS:
                     targetSpecies = formChanges[i].targetSpecies;
                     break;
+                case FORM_CHANGE_STATUS:
+                    if(GetBoxMonData(boxMon, MON_DATA_STATUS, NULL) & formChanges[i].param1)
```
Comment (Bassoonian) [inline src/pokemon.c:6236]:
```suggestion
                    if (GetBoxMonData(boxMon, MON_DATA_STATUS, NULL) & formChanges[i].param1)
```

File: data/battle_scripts_1.s [hunk 2]
```diff
@@ -6993,17 +6993,21 @@ BattleScript_CudChewActivates::
...
 	handleformchange BS_SCRIPTING, 0
 	handleformchange BS_SCRIPTING, 1
 	playanimation BS_TARGET, B_ANIM_FORM_CHANGE
 	waitanimation
 	handleformchange BS_SCRIPTING, 2
-.if B_DISGUISE_HP_LOSS >= GEN_8
-	healthbarupdate BS_SCRIPTING
-	datahpupdate BS_SCRIPTING
-.endif
+	jumpifability BS_TARGET ABILITY_DISGUISE BattleScript_ApplyDisguiseFormChangeHPLoss
```
Comment (Bassoonian) [inline data/battle_scripts_1.s:7010]:
```suggestion
	jumpifability BS_TARGET, ABILITY_DISGUISE, BattleScript_ApplyDisguiseFormChangeHPLoss
```
I'm surprised it compiles and works without the comma, but with commas are how the rest of the repo does it

Comment (TheJamesLJ) [reply data/battle_scripts_1.s:7010]: Thanks for reviewing, I'll update based on your feedback later today.

**Review by Bassoonian — APPROVED** [review]

**Pull request merged**